### PR TITLE
Nettles now inject reagents on attack

### PR DESCRIPTION
**Pull request by intrnlerr**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: 
add: Nettles now inject their reagents on attack
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Nettles can now actually be a little more helpful and not death oriented.

### CODE REVIEW

File: code/modules/hydroponics/grown/nettle.dm [hunk 1]
```diff
@@ -68,6 +68,11 @@
 
 /obj/item/reagent_containers/food/snacks/grown/nettle/afterattack(atom/A as mob|obj, mob/user,proximity)
 	. = ..()
+	if (isliving(A))
```
Comment (SpaceManiac) [inline code/modules/hydroponics/grown/nettle.dm]: this needs to go after the proximity check

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown/nettle.dm [hunk 2]
```diff
@@ -70,6 +70,11 @@
 	. = ..()
 	if(!proximity)
 		return
+	if (isliving(A))
+		var/mob/living/target = A
+		if (target.reagents)
+			var/inject_amount = rand(5, 10)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/nettle.dm]: base this off force to simulate the "leaves fall off" bit. Less Leaves = less reagent transferred?

This also needs to be checked after the `force > 0` on L77

**Comment by intrnlerr** [comment]
the nettles don't penetrate hardsuits but yeah this might not be the most balanced

**Comment by Tlaltecuhtli** [comment]
well it was intended to work like that

**Comment by ExcessiveUseOfCobblestone** [comment]
Sorta, nettles predate production genes so the intention was to transfer the appropriate acid, which it did and still does, in conjunction with the 15 force.

It was not, however, intended or balanced behind that chemical being anything available in any plant (or any chemical at all if you consider stranges). Be mindful this might see a nerf soon after implementation assuming it gets merged.

**Comment by Kagus** [comment]
I mean, on-contact nitroglycerin explosives are already just a hop skip and a jump away for botanists, I'm not sure I see how letting nettles inject reagents on melee touch in addition to how they already inject things when thrown is that big of a deal... Except for the obvious funtime generator of letting people make healing nettles that (more or less) make up for the damage caused.

**Comment by ExcessiveUseOfCobblestone** [comment]
fair enough, i'm interested in how it plays out anyways.

**Comment by optimumtact** [comment]
the fact that botany is already overpowered doesn't seem like a great justification to make it worse

**Comment by RogueSteampunker** [comment]
the fact that no one realizes this was a literal feature for the longest time and no one cared cause the abilities to abuse it were never reached a la ocram's razor makes it better

**Comment by MimicFaux** [comment]
I think the concern of ODing on healing nettles should be part of the fun. Botany is the hodgepodge workaround, the mcgyver department, the ghetto solution. You don't take a body to botany for cloning first, and you don't think of steelcaps as the first stop for metal needs. Nettles being little grown hyposprays is a new alternative that carries their own risks and rewards.

Of course like anything in spess, this can also be used for evil.

**Comment by optimumtact** [comment]
i prefer deathnettles the way they were originally.

**Pull request closed without merging**